suitesparse_graphblas/build.py

```diff
@@ -20,6 +20,13 @@
     include_dirs.append(os.path.join(sys.prefix, "Library", "include"))
     library_dirs.append(os.path.join(sys.prefix, "Library", "lib"))
 
+    # suitesparse.sh installs GraphBLAS.h here
+    include_dirs.append(os.path.join("C:\\", "Program Files (x86)", "include"))
+    # graphblas.lib and graphblas.dll.a
+    library_dirs.append(os.path.join("C:\\", "Program Files (x86)", "lib"))
+    # graphblas.dll
+    library_dirs.append(os.path.join("C:\\", "Program Files (x86)", "bin"))
```

Comment:
Can we not set these by default? Can they instead be set by changing environment variables or flag or optional argument? Not every redistribution will install to these locations when building, so I don't want to hard code these.

Reply:
Good point. CMakeLists installs to `${SUITESPARSE_LIBDIR}` (and inc/bin) but I can't find where those values are set. Maybe they aren't and those are the default paths.

I know SuiteSparse has options for setting the install location, do you know the preferred way for GraphBLAS?

Reply:
> I know SuiteSparse has options for setting the install location, do you know the preferred way for GraphBLAS?

I'm not sure I understand the question. What's the difference between `SuiteSparse` and `GraphBLAS` here?

When running cmake to build SuiteSparse:GraphBLAS, I think you can set the following variables (I'm not an expert on this):
```
CMAKE_INSTALL_PREFIX
CMAKE_INSTALL_LIBDIR
CMAKE_PREFIX_PATH
CMAKE_LIBRARY_OUTPUT_DIRECTORY
CMAKE_RUNTIME_OUTPUT_DIRECTORY
CMAKE_ARCHIVE_OUTPUT_DIRECTORY
```
This is what the conda-forge recipe for Windows does:
https://github.com/conda-forge/graphblas-feedstock/blob/main/recipe/bld.bat

When building `python-suitesparse-graphblas`, I think the usual environment variables for searching for libraries may work?

Reply:
Ah, the SuiteSparse config step has an option to set the install prefix that applies to all the libraries. You don't have to set those 5 variables individually.

The usual variables seem to work fine on Linux and macOS, but the ones you see in `build.py` are the defaults on Windows. I don't think Windows has system-wide "lib" and "include" like Unix. If you have a preference let me know.

Reply:
Gotcha. I'm happy to follow your lead here.

Perhaps we can keep what we have and address it later if there are any issues.

We'll do an alpha release first to give us an opportunity to try everything out.

Reply:
I'm testing a rebase that  installs to "C:\GraphBLAS". Feels cleaner and doesn't make path assumptions.

I think Alpha is a good idea. Would you prefer if this PR didn't override the existing `wheel.yml` and instead made an independent one? That way you don't have to commit to this one, and you can switch from Test PyPI when you're comfortable.

We can also coordinate about how to delete the Test PyPI. I think it auto deletes after 30 days, but I can delete earlier so you can claim the package ID. You have to make another account and API token, they don't share with regular PyPI. I uploaded mostly as a proof that this workflow works.

Reply:
> I'm testing a rebase that installs to "C:\GraphBLAS". Feels cleaner and doesn't make path assumptions.

Cool, sounds great.

I say override `wheel.yml`. I'm also happy to let you "own" Test PyPI ;) . Can you try adding me (`eriknw`) to the project?

I suppose we should probably do what `scikit-learn` does in their `publish_pypi.yml` which is to have a workflow variable to control to which repo to upload to enable easier testing. Up to you. I think our normal workflow will be for wheels to automatically build and upload to PyPI whenever we to a tag/release, and we'll probably do alpha releases when anything with the build system changes.